packages/server/lib/util/ci_provider.js

```diff
@@ -222,7 +222,27 @@ const _providerCiParams = () => {
       'PULL_REQUEST_BASE_BRANCH', // Name of the branch that the pull request will be merged into. It should be the same as BASE_BRANCH.
       'PULL_REQUEST_REPO_FULL_NAME', // Full name of the repository from where the pull request originated.
     ]),
-    snap: null,
+    goCD: extract([
+      'GO_SERVER_URL',
+      'GO_ENVIRONMENT_NAME',
+      'GO_PIPELINE_NAME',
+      'GO_PIPELINE_COUNTER',
+      'GO_PIPELINE_LABEL',
+      'GO_STAGE_NAME',
+      'GO_STAGE_COUNTER',
+      'GO_JOB_NAME',
+      'GO_TRIGGER_USER',
+      'GO_DEPENDENCY_LABEL_${pipeline_name}',
```

Comment:
hmm, are these dynamically set?

Reply:
Sorry for late reply, I have busy work that need to handle it first. So, we need to change it into **GO_DEPENDENCY_LABEL** ?. Currently, there are no example, to write dynamic variable.